src/components/offer/offer.tsx

```diff
@@ -0,0 +1,121 @@
+                </svg>
+                <span className="visually-hidden">To bookmarks</span>
+              </button>
+            </div>
+            <div className="offer__rating rating">
+              <div className="offer__stars rating__stars">
+                <span style={{width: offer?.rating}}></span>
+                <span className="visually-hidden">Rating</span>
+              </div>
+              <span className="offer__rating-value rating__value">{offer?.rating}</span>
+            </div>
+            <ul className="offer__features">
+              <OfferFeature textContent={ offer?.city.name } />
+              <OfferFeature textContent= {`${offer?.bedrooms} Bedrooms`} />
+              <OfferFeature textContent= {`Max ${ offer?.maxAdults } adults`} />
+            </ul>
+            <div className="offer__price">
+              <b className="offer__price-value">&euro;{offer?.price}</b>
+              <span className="offer__price-text">&nbsp;night</span>
+            </div>
+            <div className="offer__inside">
+              <h2 className="offer__inside-title">What&apos;s inside</h2>
+              <ul className="offer__inside-list">
+                <OfferInside textContent='Wi-Fi'/>
+                <OfferInside textContent='Washing machine'/>
+                <OfferInside textContent='Towels'/>
+                <OfferInside textContent='Heating'/>
+                <OfferInside textContent='Coffee machine'/>
+                <OfferInside textContent='Baby seat'/>
+                <OfferInside textContent='Kitchen'/>
+                <OfferInside textContent='Dishwasher'/>
+                <OfferInside textContent='Cabel TV'/>
+                <OfferInside textContent='Fridge'/>
+              </ul>
+            </div>
+            <div className="offer__host">
+              <h2 className="offer__host-title">Meet the host</h2>
+              <OfferHost image={ offer?.host.avatarUrl } name={ offer?.host.name } status={ offer?.host.isPro } description={ offer?.description }/>
+            </div>
+            <section className="offer__reviews reviews">
+              <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">1</span></h2>
+              <ul className="reviews__list">
+                <OfferReview image='img/avatar-max.jpg' userName='Max' width='80%' userText='A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.' date='2019-04-24' monthYear='' />
+              </ul>
+              <form className="reviews__form form" action="#" method="post">
+                <label className="reviews__label form__label" htmlFor="review">Your review</label>
+                <div className="reviews__rating-form form__rating">
+                  <OfferStar value='5' id='5-stars' title='perfect' />
+                  <OfferStar value='4' id='4stars' title='good' />
```

Comment:
Говоря про важность циклов. Если автоматизировать работу ручную, то таких ошибок не случается

```suggestion
                  <OfferStar value='4' id='4-stars' title='good' />
```